mica-core/src/main/java/org/obiba/mica/micaConfig/service/MicaConfigService.java

```diff
@@ -265,8 +269,13 @@ public String encrypt(String plain) {
   }
 
   public String decrypt(String encrypted) {
-    ByteSource decrypted = cipherService.decrypt(Hex.decode(encrypted), getSecretKey());
-    return CodecSupport.toString(decrypted.getBytes());
+    try {
+      ByteSource decrypted = cipherService.decrypt(Hex.decode(encrypted), getSecretKey());
+      return CodecSupport.toString(decrypted.getBytes());
+    } catch (CryptoException e) {
+      logger.warn(String.format("Someone tried to use an invalid key [%s]", encrypted));
```

Comment:
It's not possible. This case is when a user with shareKey try to access to a resource. So user is not loggedIn and shareKey is not "decryptable". Si impossible to get the person which create the key.